doc/usage/restructuredtext/domains.rst

```diff
@@ -50,13 +50,18 @@ give the directive option flag ``:noindexentry:``.
 If you want to typeset an object description, without even making it available
 for cross-referencing, you can give the directive option flag ``:noindex:``
 (which implies ``:noindexentry:``).
+If you do not want to typeset anything but create a target and index entry for
+later reference, you can give the directive option flag ``:notypesetting:``.
```

Comment:
> This may be slightly ambiguous: does notypesetting imply not noindex/noindex? I would assume that it is independent. I.e., if the user specifies notypesetting and noindexentry/noindex, then there will be almost no side-effect of having the directive.

You are correct in your assumption: Currently, `:notypesetting:` and `:noindex:`/`:noindexentry:` are completely independent, neither implies the other and they can freely mixed.
The combination `:notypesetting:` and `:noindex:` results in essentially an empty `<index>` node which results in a noop for the usual outputs. However, some domain out there might provide ids in the nodes it produces even if `:noindex:` is provided. In this case we generate a `<target>` node with those ids so they'll become linkable.
The combination `:notypesetting:` and `:noindexentry:` usually results in an linkable `<target>` node, i.e. can be referenced later but as suggested no entry on the generated index page.

I'll use your wording suggestions once I know what to do with the `option_spec`.